adsingestp/parsers/jats.py

```diff
@@ -146,6 +146,10 @@ def _match_xref_clean(self):
 
                 if auth.get("email", []):
                     auth["email"] = self._fix_email(auth["email"])
+                # the auth["email"] list may not maintain order when
+                # parsing the same file.  For consistency, sort it.
+                if type(auth["email"]) != str and type(auth["email"]) == list:
+                    auth["email"].sort()
```

Comment:
I've done a lot more poking on this today and I don't think the random ordering behavior you're seeing during testing could come from this code block. In the example XML you give, the emails are stored in `author-notes`, outside of the `contrib-group` blocks. These are added to the `self.xref_dict` in the block that starts on L348, and then mapped into the correct author in the block starting on L140. However, the item in that dictionary is the full unparsed author note:
```
print(self.xref_dict['cor1'])
"To whom correspondence may be addressed. Email:  dr.guoqihao@126.com  or  boip@ust.hk ."```
In L148, this gets passed to `self._fix_email`, which will extract the two email addresses from that string and turn them into an array. However, in that subfunction, L67 initializes the list of emails as a set, not a list. Sets are unordered. The set is converted to a list for output, but too late, the randomness is already there.

SO. To fix this, scrap the sort here and implement a fix in `self._fix_email` instead. Instead of initializes as a set, either initialize as a dict (dicts are sorted by default in... 3.6? onwards) with the emails as the keys (no values) (see this SO answer for an idea: https://stackoverflow.com/questions/1653970/does-python-have-an-ordered-set). Or there's a library to implement ordered sets (https://www.udacity.com/blog/2021/11/python-ordered-sets-an-overview.html). Either option will work.

You can also scrap the sorting in L166 for the ORCID IDs, and fix the `self._fix_orcid` function in a similar way - it's also initialized using a set.